### PR TITLE
# 131356237 user chat

### PR DESCRIPTION
**Pull request by andela-cnwankwo**
#### What does this PR do?

Implement group chat feature.

#### Description of Task to be completed?

Users should be able to chat in a group while they are playing the game.

#### How should this be manually tested?

https://hyuga-cfh-staging-pr-30.herokuapp.com/

#### Any background context you want to provide?

N/A
#### What are the relevant pivotal tracker stories?

#131356237 Users should be able to chat with the group while they are in a game

#### Screenshots (if appropriate)

N/A

#### Questions:

N/A

### CODE REVIEW

File: public/views/chat.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <div id="messages-card" class="mdl-card mdl-shadow--2dp mdl-cell mdl-cell--12-col mdl-cell--6-col-tablet mdl-cell--6-col-desktop">
+        <div id="chat_header" ng-click="resize()">Chat<span id="unread" class="mdl-badge" data-badge=""></span></div>
+        <div id="show_message" class="mdl-card__supporting-text mdl-color-text--grey-600">
+          <div id="messages">
+            <span id="message-filler"></span>
+          </div>
+          <form id="message-form">
+            <div class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">
+              <input class="mdl-textfield__input" type="text" id="message">
+              <label class="mdl-textfield__label" for="message">Message...</label>
```
Comment (andela-oagunbiade) [inline public/views/chat.html]: LGTM. You could howevere, have the "Message" label set as a placeholder instead.  It

Comment (andela-cnwankwo) [inline public/views/chat.html]: Ok, I will do that.

File: app/views/includes/foot.jade [hunk 2]
```diff
@@ -44,5 +45,22 @@ script(type='text/javascript', src='/js/init.js')
 //Socket.io Client Library
 script(type='text/javascript', src='/socket.io/socket.io.js')
 
+// firebase
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-app.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-database.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-messaging.js')
+script(type='text/javascript', defer src="https://code.getmdl.io/1.1.3/material.min.js")
```
Comment (andela-cnwankwo) [inline app/views/includes/foot.jade]: I will go ahead and remove the mdl script tags since we are using materialize now.

File: app/views/includes/foot.jade [hunk 3]
```diff
@@ -44,5 +45,20 @@ script(type='text/javascript', src='/js/init.js')
 //Socket.io Client Library
 script(type='text/javascript', src='/socket.io/socket.io.js')
 
+// firebase
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-app.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-database.js')
+script(type='text/javascript', src='https://www.gstatic.com/firebasejs/3.5.0/firebase-messaging.js')
+script.
+  var config = {
```
Comment (neddinn) [inline app/views/includes/foot.jade]: Not a good practise to have the config files here. Should be stored somewhere in a config service.

File: app/controllers/api/users.js [hunk 4]
```diff
@@ -63,3 +63,11 @@ exports.signup = function (req, res) {
     });
   });  
 };
+
+exports.currentUser = function (req, res) {
+  console.log(req.user);
```
Comment (andela-koni) [inline app/controllers/api/users.js]: remove console.log

File: app/controllers/api/users.js [hunk 5]
```diff
@@ -63,3 +63,11 @@ exports.signup = function (req, res) {
...
+  if(req.user){
+   return res.send(req.user);
+  }
+  res.status(404).send('Not Found!');
+};
```
Comment (andela-koni) [inline app/controllers/api/users.js]: I thought we added the newline prompt to our linter, please add a new line at the end of the file.

File: app/views/includes/head.jade [hunk 6]
```diff
@@ -23,16 +23,15 @@ head
   meta(property='og:site_name', content='Cards for Humanity')
   meta(property='fb:admins', content='APP_ADMIN')
 
-  //-
   link(rel='stylesheet', href='/lib/bootstrap/dist/css/bootstrap.min.css')
-  link(rel='stylesheet', href='/css/common.css')
+    link(rel='stylesheet', href='/css/common.css')
```
Comment (andela-koni) [inline app/views/includes/head.jade]: why did you add a space? please indent properly

File: app/views/index.jade [hunk 7]
```diff
@@ -2,4 +2,4 @@ extends layouts/default
 block content
   section(data-ng-view, id = "main-container")
   script(type="text/javascript").
-    window.user = !{user};
+    localStorage.user = JSON.stringify(!{user});
```
Comment (andela-koni) [inline app/views/index.jade]: why are you negating the user here?

File: config/apiRoutes.js [hunk 8]
```diff
@@ -6,4 +6,5 @@ module.exports = function(app) {
   // User routes
   app.post('/api/auth/signup', users.signup);
   app.post('/api/auth/login', users.login);
+  app.get('/api/auth/currentuser', users.currentUser);
```
Comment (andela-koni) [inline config/apiRoutes.js]: `currentuser` should be camelCase or snake_case preferably snake_case. Please change this.

File: public/js/controllers/game.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
 angular.module('mean.system')
-  .controller('GameController', ['$scope', 'game', '$timeout', '$location', 'MakeAWishFactsService', '$dialog', function ($scope, game, $timeout, $location, MakeAWishFactsService) {
+  .controller('GameController', ['$scope', 'game', 'chat', '$timeout', '$location', 'MakeAWishFactsService', '$dialog', function($scope, game, chat, $timeout, $location, MakeAWishFactsService, $dialog) {
```
Comment (andela-koni) [inline public/js/controllers/game.js]: Break this into multiple lines
